hazelcast-client/src/main/java/com/hazelcast/client/proxy/ClientMapProxy.java

```diff
@@ -538,16 +552,20 @@ public V put(K key, V value, long ttl, TimeUnit timeunit) {
         checkNotNull(key, NULL_KEY_IS_NOT_ALLOWED);
         checkNotNull(value, NULL_VALUE_IS_NOT_ALLOWED);
 
-        return putInternal(ttl, timeunit, DEFAULT_MAX_IDLE, MILLISECONDS, key, value);
+        return putInternal(ttl, timeunit, null, null, key, value);
     }
 
-    protected V putInternal(long ttl, TimeUnit ttlUnit, long maxIdle, TimeUnit maxIdleUnit, Object key, Object value) {
+    protected V putInternal(long ttl, TimeUnit ttlUnit, Long maxIdle, TimeUnit maxIdleUnit, Object key, Object value) {
         Data keyData = toData(key);
         Data valueData = toData(value);
         long ttlMillis = getTimeInMillis(ttl, ttlUnit);
-        long maxIdleMillis = getTimeInMillis(maxIdle, maxIdleUnit);
-        ClientMessage request = MapPutCodec.encodeRequest(name, keyData, valueData,
-                getThreadId(), ttlMillis, maxIdleMillis);
+        ClientMessage request;
+        if (maxIdle != null) {
+            request = MapPutCodec.encodeRequest(name, keyData, valueData,
+                    getThreadId(), ttlMillis, getTimeInMillis(maxIdle, maxIdleUnit));
```

Comment:
Looks like the client and the member proxy implementations diverge if the `maxIdleUnit` is `null`.
Client interprets `maxIdle` as milliseconds in this case, while `MapProxySupports` throws NPE.
Same with `ttlUnit` btw.

This needs `null` timeUnits to be passed on the interface methods though. Is this expected?

Reply:
Good point. I think we should remove null checks should not allow null to be passed as on the member side.